modules/basegl/src/processors/imageprocessing/imagelayoutgl.cpp

```diff
@@ -166,58 +200,87 @@ void ImageLayoutGL::propagateEvent(Event* event, Outport* source) {
 
-    leftMinMax_.setVisible(false);
-    rightMinMax_.setVisible(false);
-    topMinMax_.setVisible(false);
-    bottomMinMax_.setVisible(false);
+    bool boundsVisible = true;
+    bool lVisible = false;  // leftMinMax_
+    bool rVisible = false;  // rightMinMax_
+    bool tVisible = false;  // topMinMax_
+    bool bVisible = false;  // bottomMinMax_
 
     switch (layout_.getSelectedValue()) {
         case Layout::HorizontalSplit:
-            horizontalSplitter_.setVisible(true);
-            topMinMax_.setVisible(true);
-            bottomMinMax_.setVisible(true);
+            hVisible = true;
+            tVisible = true;
+            bVisible = true;
+
             break;
         case Layout::VerticalSplit:
-            verticalSplitter_.setVisible(true);
-            leftMinMax_.setVisible(true);
-            rightMinMax_.setVisible(true);
+            vVisible = true;
+            lVisible = true;
+            rVisible = true;
             break;
         case Layout::CrossSplit:
-            horizontalSplitter_.setVisible(true);
-            verticalSplitter_.setVisible(true);
-            leftMinMax_.setVisible(true);
-            rightMinMax_.setVisible(true);
-            topMinMax_.setVisible(true);
-            bottomMinMax_.setVisible(true);
+            hVisible = true;
+            vVisible = true;
+
+            lVisible = true;
+            rVisible = true;
+            tVisible = true;
+            bVisible = true;
             break;
         case Layout::ThreeLeftOneRight:
-            vertical3Left1RightSplitter_.setVisible(true);
-            leftMinMax_.setVisible(true);
-            rightMinMax_.setVisible(true);
+            v3LeftVisible = true;
+            lVisible = true;
+            rVisible = true;
             break;
         case Layout::ThreeRightOneLeft:
-            vertical3Right1LeftSplitter_.setVisible(true);
-            leftMinMax_.setVisible(true);
-            rightMinMax_.setVisible(true);
+            v3RightVisible = true;
+            lVisible = true;
+            rVisible = true;
             break;
         case Layout::HorizontalSplitMultiple:
         case Layout::VerticalSplitMultiple:
         case Layout::Single:
         default:
+            boundsVisible = false;
             break;
     }
 
+    horizontalSplitter_.setVisible(hVisible);
+    verticalSplitter_.setVisible(vVisible);
+    vertical3Left1RightSplitter_.setVisible(v3LeftVisible);
+    vertical3Right1LeftSplitter_.setVisible(v3RightVisible);
+
+    bounds_.setVisible(boundsVisible);
+    leftMinMax_.setVisible(lVisible);
+    rightMinMax_.setVisible(rVisible);
+    topMinMax_.setVisible(tVisible);
+    bottomMinMax_.setVisible(bVisible);
+
     if (propagate) {
         ResizeEvent e(currentDim_);
         propagateEvent(&e, &outport_);
     }
 }
 
+bool ImageLayoutGL::isConnectionActive(Inport* from, Outport* to) const {
+    IVW_ASSERT(from == &multiinport_, "only one inport");
```

Comment:
I do not get this check, is it not allowed to have only one input? That will happen for sure when setting up a new network.

Reply:
This processor only has one inport. Perhaps it should say only one inport though

Reply:
Ah, it does :D

Reply:
Why do we need this test? Is it if someone calls it themself from somewhere? In that case I guess it should say something like "isConnectionActive called with an inport not belling to current processor"

Reply:
Because the code below assumes that the supplied inport is the multiinport_. Want to make sure that this assumption is valid later even if someone decides to alter the processor.

Reply:
Then use a propper error message